webapp/src/utils/adapters/index.ts

```diff
@@ -27,3 +31,51 @@ export interface IMessageAdapter {
 export interface ITranslationAdapter {
   getTranslations(): Promise<TranslationMap>;
 }
+
+/** convert { [id]: { file, texts } } to { [id]: text } */
+export function dehydrateMessageMap(
+  translations: MessageMap,
+): Record<string, string> {
+  const output: Record<string, string> = {};
+  Object.entries(translations).forEach(([id, { text }]) => {
+    output[id] = text;
+  });
+  return output;
+}
+
+type TranslationFilenameMap = Record<
+  string, // lang
+  Record<
+    string, // filename
+    Record<
+      string, // msg Id
+      string // text
+    >
+  >
+>;
+
+/** convert TranslateMap to { [lang]: { [filename]: {[msgId]: text} } */
+export function groupByFilename(
+  translationMap: TranslationMap,
+): TranslationFilenameMap {
+  const output: TranslationFilenameMap = {};
+  Object.entries(translationMap).forEach(([lang, messageMap]) => {
+    Object.entries(messageMap).forEach(([msgId, { text, sourceFile }]) => {
+      if (!output[lang]) {
+        output[lang] = {};
+      }
+      if (!output[lang][sourceFile]) {
+        output[lang][sourceFile] = {};
+      }
+      const shortId = getShortId(msgId, sourceFile);
+      output[lang][sourceFile][shortId] = text;
+    });
+  });
+  return output;
+}
+
+function getShortId(msgId: string, sourceFile: string): string {
+  const sourceFileDotCount = sourceFile.replace('/', '.').split('.').length - 1;
+  const msgIdArray = msgId.split('.').splice(sourceFileDotCount);
```

Comment:
This code overestimates how much of the message id comes from the path. The filename --- the last segment of the path, typically contains a dot, for example in 'sv.yml' but can in theory contain more dots too, for example 'sv.skånska.yaml'.

For example `getShortId('a.b.c', 'a/sv.yml')` should return 'b.c' but this implementation will return just 'c'.